src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -125,67 +121,90 @@
             (some (get rule-params k)
                   (flatten [(get request-params (name k))])))))
 
-(schema/defn match? :- RuleMatch
+(schema/defn sort-rules :- [Rule]
+  "Sorts the rules based on their :sort-order, and then their :name if they
+   have the same sort order value."
+  [rules :- [Rule]]
+  (sort-by (juxt :sort-order :name) rules))
+
+(schema/defn requestor :- schema/Str
+  "Returns a string that identifies the source of the request containing
+   at least the IP address and the hostname if available."
+  [request :- ring/Request]
+  (if-let [name (ring/authorized-name request)]
+    (format "%s(%s)" name (:remote-addr request))
+    (str (:remote-addr request))))
```

Comment:
This function is where we need to differentiate between a `:name nil` and `:name ""`.  The `(format ...)` line was extracted from `request->description` and was incorrectly returning the `(1.2.3.4)` when it should've been returning `1.2.3.4` in the case of no request `:name`.

It was incorrectly doing this because `ring/set-authorization-name` was converting the input from `nil`->`""`.

Anyway, much ado about nothing, just explaining why I got rid of the `nil`<->`""` conversion